spinalcordtoolbox/download.py

```diff
@@ -0,0 +1,177 @@
+              'https://www.neuro.polymtl.ca/_media/downloads/sct/20191029_PAM50.zip'],
+    'MNI-Poly-AMU': ['https://osf.io/sh6h4/?action=download',
+                     'https://www.neuro.polymtl.ca/_media/downloads/sct/20170310_MNI-Poly-AMU.zip'],
+    'gm_model': ['https://osf.io/ugscu/?action=download',
+                 'https://www.neuro.polymtl.ca/_media/downloads/sct/20160922_gm_model.zip'],
+    'optic_models': ['https://osf.io/g4fwn/?action=download',
+                     'https://www.neuro.polymtl.ca/_media/downloads/sct/20170413_optic_models.zip'],
+    'pmj_models': ['https://osf.io/4gufr/?action=download',
+                   'https://www.neuro.polymtl.ca/_media/downloads/sct/20170922_pmj_models.zip'],
+    'binaries_debian': ['https://osf.io/bt58d/?action=download',
+                        'https://www.neuro.polymtl.ca/_media/downloads/sct/20190930_sct_binaries_linux.tar.gz'],
+    'binaries_centos': ['https://osf.io/8kpt4/?action=download',
+                        'https://www.neuro.polymtl.ca/_media/downloads/sct/20190930_sct_binaries_linux_centos6.tar.gz'],
+    'binaries_osx': ['https://osf.io/msjb5/?action=download',
+                     'https://www.neuro.polymtl.ca/_media/downloads/sct/20190930_sct_binaries_osx.tar.gz'],
+    'course_hawaii17': 'https://osf.io/6exht/?action=download',
+    'course_paris18': ['https://osf.io/9bmn5/?action=download',
+                       'https://www.neuro.polymtl.ca/_media/downloads/sct/20180612_sct_course-paris18.zip'],
+    'course_london19': ['https://osf.io/4q3u7/?action=download',
+                        'https://www.neuro.polymtl.ca/_media/downloads/sct/20190121_sct_course-london19.zip'],
+    'course_beijing19': ['https://osf.io/ef4xz/?action=download',
+                         'https://www.neuro.polymtl.ca/_media/downloads/sct/20190802_sct_course-beijing19.zip'],
+    'deepseg_gm_models': ['https://osf.io/b9y4x/?action=download',
+                          'https://www.neuro.polymtl.ca/_media/downloads/sct/20180205_deepseg_gm_models.zip'],
+    'deepseg_sc_models': ['https://osf.io/avf97/?action=download',
+                          'https://www.neuro.polymtl.ca/_media/downloads/sct/20180610_deepseg_sc_models.zip'],
+    'deepseg_lesion_models': ['https://osf.io/eg7v9/?action=download',
+                              'https://www.neuro.polymtl.ca/_media/downloads/sct/20180613_deepseg_lesion_models.zip'],
+    'c2c3_disc_models': ['https://osf.io/t97ap/?action=download',
+                         'https://www.neuro.polymtl.ca/_media/downloads/sct/20190117_c2c3_disc_models.zip']
+    }
+
+
+def download_data(urls):
+    """Download the binaries from a URL and return the destination filename
+
+    Retry downloading if either server or connection errors occur on a SSL
+    connection
+    urls: list of several urls (mirror servers) or single url (string)
+    """
+
+    # if urls is not a list, make it one
+    if not isinstance(urls, (list, tuple)):
+        urls = [urls]
```

Comment:
This should be a bit more reliable:

```suggestion
    # if urls is not a list, make it one
    if isinstance(urls, str):
        urls = [urls]
```

Reply:
indeed-- fixed in 88a04b56184f5d772dcaeaff18fbbeb18a704f4d